runtime/recipe-index.js

```diff
@@ -136,6 +136,11 @@ export class RecipeIndex {
 
     let results = [];
     for (let recipe of this._recipes) {
+      if (recipe.particles.some(particle => !particle.name)) {
+        // Skip recipes where not all verbs are resolved to specific particles
+        // to avoid trying to coalesce a recipe with itself.
+        continue;
+      }
```

Comment:
Does this reduce the search space? If A could coalesce with B but B's verbs aren't resolved, could there be a situation where B can't coalesce with C but A+B can?

Reply:
if A can coalesce with B then it'll coalesce after B's verbs are resolved (note that in order for coalescing to run A will be a terminal recipe already). 
there is an attempt (below) to compare particle names in both coalesced recipes to avoid situation when recipe is coalesced with itself, but it doesn't work properly if we consider recipes with unresolved verbs.